feathr_project/feathr/utils/platform.py

```diff
@@ -0,0 +1,38 @@
+"""Platform utilities.
+Refs: https://github.com/microsoft/recommenders/blob/main/recommenders/utils/notebook_utils.py
+"""
+from pathlib import Path
+
+
+def is_jupyter() -> bool:
+    """Check if the module is running on Jupyter notebook/console.
+
+    Returns:
+        bool: True if the module is running on Jupyter notebook or Jupyter console, False otherwise.
+    """
+    try:
+        shell_name = get_ipython().__class__.__name__
+        if shell_name == "ZMQInteractiveShell":
```

Comment:
Sounds this is hard coded? `ZMQInteractiveShell `

Reply:
This is a bit hacky but works. Ref is: https://stackoverflow.com/questions/15411967/how-can-i-check-if-code-is-executed-in-the-ipython-notebook. I'll add the reference to the function docstring too.

Pre-loaded module `get_ipython()` tells you whether you are running inside ipython or not, and `ZMQInteractiveShell` tells you if this is an interactive mode (notebook).

Reply:
@loomlike interesting to learn that. Can you put that in the comments so others can follow?